### PR TITLE
Deploy a Github Action to auto-update gh-pages branch

### LINKED ISSUE #149: Evaluate the strategy for deploying docs
This issue can be closed when someone has:

- [x] contemplated the proper structure for holding the doxygen docs (probably source files only in devel & master, built files only in gh-pages branch)
- [x] Create a gh-pages branch (orphan it)
- [x] Extract the docs subfolder, and all of its history, into the new gh-pages branch
- [x] Update settings in the repo options to deploy pages at gh-pages rather than master/docs (ask @katyhuff to do it)

### PR DESCRIPTION
**Pull request by smpark7**
Closes #149 

The `gh-pages` branch currently holds the built Doxygen files for Moltres' documentation on `arfc.github.io/moltres`. This pull request deploys a Github Action workflow which automatically updates the built Doxygen files in the `gh-pages` branch.

This workflow triggers on any push or pull request into the `master` branch. The sequence of events is as follows:
1. A pull request is approved and merged into `devel`
2. `moosetest` pushes latest commits from `devel` to `master` after CIVET's continuous integration testing
3. This push into `master` triggers this workflow
4. The `mattnotmitt/doxygen-action@v1` action builds Doxygen files based on the settings in `$MOLTRES/Doxyfile`. Doxygen creates a `docs/` subdirectory to hold the built files.
5. The `peaceiris/actions-gh-pages@v3` action compares the new built files in `docs/` with the existing built files in `gh-pages`. If there are no changes (i.e. empty commit), it terminates. If there are changes, it pushes the changes to `gh-pages`.

@munkm may have to redeploy Github Pages as described in this [link](https://github.com/marketplace/actions/github-pages-action#%EF%B8%8F-first-deployment-with-github_token) after merging this PR if it breaks the existing Github Pages functionality.

### CODE REVIEW

File: .github/workflows/auto-update-gh-pages.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# This GitHub Action automatically updates built Doxygen files for Moltres' documentation
+# on `https://arfc.github.io/moltres/` every time a new pull request is merged into devel
+# and subsequently into master after CIVET's continuous integration tests.
+
+name: Doxygen & Github Pages Action
+
+# Activate on pushes and pull requests into master
+on:
+  push:
+    branches: [ master ]
```
Comment (munkm) [inline .github/workflows/auto-update-gh-pages.yml:10]: should these be `devel` and not master?

File: .github/workflows/auto-update-gh-pages.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  pull_request:
+    branches: [ master ]
```
Comment (munkm) [inline .github/workflows/auto-update-gh-pages.yml:12]: same here

**Review by munkm — COMMENTED** [review]
I noticed that there's a `gh-pages` branch (last updated in Jan) for moltres and then a `gh-action-for-gh-pages` branch updated a couple of weeks ago. Is that intentional, and was it generated by your local CI somehow? Can we limit it all to just the `gh-pages` branch?

File: .github/workflows/auto-update-gh-pages.yml (hunk 1 above)
Comment (smpark7) [reply .github/workflows/auto-update-gh-pages.yml:10]: I was actually not sure which branch to track. On one hand, our pull requests merge directly in `devel`. On the other hand, INL's integration testing tool CIVET runs more comprehensive testing on the `devel` branch before it pushes to `master`.

**Comment by smpark7** [comment]
Sorry for the confusion from the branch names. `gh-pages` branch is the orphan branch that holds the built html Doxygen files for `arfc.github.io/moltres`. `gh-action-for-gh-pages` is not an orphan branch. `gh-action-for-gh-pages` contains the commits for this pull request.

The reason I created this branch on `arfc/moltres` instead of my own fork is because GitHub Actions didn't let me do that. It only gave me the option of 1) directly pushing this commit into `devel` without a pull request or 2) creating this branch on `arfc/moltres` for a pull request.

**Review by munkm — APPROVED** [review]

**Pull request merged**